dnf/transaction_sr.py

```diff
@@ -309,7 +320,7 @@ def _replay_pkg_action(self, pkg_data):
                 query = query_repo.union(query.installed())
 
         if not query:
-            self._raise_or_warn(self._skip_unavailable, _('Cannot find rpm nevra "{nevra}".').format(nevra=nevra))
+            self._raise_or_warn(self._skip_unavailable, PackageNotFoundError(nevra))
```

Comment:
I don't like this. You're going the path of least resistance, but now you're putting two kinds of objects into the `self._warnings` list and that makes the code unnecessarily more complicated. You can no longer access the `_warnings` members uniformly. And the new `print_error()` method is kind of just a hack...

If you want to go this way, please update the rest of the code to always put the exception classes into the `_warnings`, but I don't like to do that just for the reason of putting a single different exception for this case in there. I still think the rest of the transaction store/replay messages could be adjusted to make them uniform with the `"No package %s available."`. If we want to preserve that message across different dnf commands, then we should stick to the formulation and keep all the messages somewhat consistent. If you want I can do the work of updating the messages.